### ISSUE TITLE
Unlimited lifetime for existing technologies

### ISSUE THREAD

**Issue by augbra** [EnergyModelsX/EnergyModelsInvestments.jl#30]
If an investment is defined with a lifetime, this lifetime is only applied when the technology is invested in. If the technology already exists at the start, it has unlimited lifetime.

**Comment by JulStraus** [comment]
Yes, this is true. That is mostly related to the initial design choices within the project CleanExport as we did not have existing technologies with an end of life. We are currently investigating a few approaches regarding a change in the calculation of the lifetime, including cost for retiring in [Issue 26](https://github.com/EnergyModelsX/EnergyModelsInvestments.jl/issues/26).

We are however aware that we currently do not highlight this sufficiently well in the documentation. It will be included  within [Pull request 28](https://github.com/EnergyModelsX/EnergyModelsInvestments.jl/pull/28) as said PR already has significant changes in the documentation.

**Comment by JulStraus** [comment]
Thinking a bit further about it, it is still possible to have a retiring capacity in the current design, although this requires a bit of tweaking.

Consider you have a source node with a reduced capacity in the individual strategic periods given as
```julia
# Create the investment data for the source node
investment_data_source = SingleInvData(
    FixedProfile(300 * 1e3),  # capex [€/MW]
    FixedProfile(30),       # max installed capacity [MW]
    ContinuousInvestment(FixedProfile(0), FixedProfile(30)),
    # Line above: Investment mode with the following arguments:
    # 1. argument: min added capactity per sp [MW]
    # 2. argument: max added capactity per sp [MW]
    lifemode(20),     # Lifetime mode
)

# Create the source
source = RefSource(
    "electricity source",       # Node ID
    StrategicProfile([10, 5, 0]),  # Capacity in MW
    FixedProfile(10),           # Variable OPEX in EUR/MW
    FixedProfile(5),            # Fixed OPEX in EUR/year
    Dict(Power => 1),           # Output from the Node, in this gase, Power
    [investment_data_source],   # Additional data used for adding the investment data
)
```

In this situation, we do not really see the impact of the reduced capacity. One alternative to still keep it present would be given by creating two source nodes, one with investment data and without capacity and one without investment data and the declining capacity:

```julia
# Create the source node with declining capacity
source_1 = RefSource(
    "electricity source 1",       # Node ID
    StrategicProfile([10, 5, 0]),  # Capacity in MW
    FixedProfile(10),           # Variable OPEX in EUR/MW
    FixedProfile(5),            # Fixed OPEX in EUR/year
    Dict(Power => 1),           # Output from the Node, in this gase, Power
)

# Create the source node with investments
source_2 = RefSource(
    "electricity source 2",       # Node ID
    FixedProfile(0),            # Capacity in MW
    FixedProfile(10),           # Variable OPEX in EUR/MW
    FixedProfile(5),            # Fixed OPEX in EUR/year
    Dict(Power => 1),           # Output from the Node, in this gase, Power
    [investment_data_source],   # Additional data used for adding the investment data
)
```

This approach has the disadvantages of

1. introducing more variables and
2. require the introduction of more links.

It is however an equivalent approach for incorporating an existing capacity with a limited lifetime.

**Comment by JulStraus** [comment]
Revisiting this issue, I personally would avoid the solution outlined in my last comment. A major problem here is that we do not allow in this situation any longer the direction inclusion of strategic uncertainty as incorporated in `TimeStruct` though *[Version 0.9](https://github.com/sintefore/TimeStruct.jl/releases/tag/v0.9.0).

I have to come up with a new solution to the given problem.
